### PR TITLE
feat: default session id in frontend api

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

What problem are we solving?
* enabled info and variant info in the same feature have to be consistent when stickiness is set to default
* also we want to have consistency between parent and child features 

Solution:
* if the user didn't provide sessionId, generate one one the fly

## Discussion points

Is Math.random() good enough?

### CODE REVIEW

File: src/lib/services/proxy-service.ts
Comment (kwasniew) [reply src/lib/services/proxy-service.ts:77]: we're not doing cryptography here

File: src/lib/services/proxy-service.ts
Comment (kwasniew) [inline src/lib/services/proxy-service.ts:86]: we don't have to use forceGetVariant anymore. Since we have sessionId we don't have to care how getVariant is implemented locally.

**Review by sighphyre — APPROVED** [review]
Oh awesome! This LGTM

Guessing a test is borderline impossible here? I think that's okay, this is simple enough

**Comment by kwasniew** [comment]
@sighphyre I can think of a property based test or run a regular test 50 times and verify some properties. It will be more work than the change itself but maybe it's worth the effort.

**Comment by sighphyre** [comment]
> @sighphyre I can think of a property based test or run a regular test 50 times and verify some properties. It will be more work than the change itself but maybe it's worth the effort.

Yeah, it may be a lot of work. I also think it may be worth it. A prop test would be awesome. We've done some "run this 10K times and poke at the statistics" tests before, they have been useful but they have drawbacks

Either way, I'm not going to block this for not having one of those. I still think LGTM, I'll leave it up to you if you want to that now, later or never

**Review by Tymek — APPROVED** [review]

**Pull request merged**
